rust/core-lib/src/styles.rs

```diff
@@ -22,7 +22,7 @@ use syntect::highlighting::{Color, Theme, ThemeSet, Highlighter};
 
 pub use syntect::highlighting::ThemeSettings;
 
-const N_RESERVED_STYLES: usize = 2;
+const N_RESERVED_STYLES: usize = 8;
```

Comment:
I feel like style 2 is hardcoded somewhere in the word wrap code, maybe take a quick look? It might need to be updated.